rs/ledger_suite/common/ledger_core/src/approvals.rs

```diff
@@ -73,14 +61,10 @@ pub trait AllowancesData {
         &mut self,
     ) -> Option<((Self::AccountId, Self::AccountId), Allowance<Self::Tokens>)>;
 
-    fn oldest_arrivals(&self, n: usize) -> Vec<(Self::AccountId, Self::AccountId)>;
-
     fn len_allowances(&self) -> usize;
 
     fn len_expirations(&self) -> usize;
 
-    fn len_arrivals(&self) -> usize;
-
     fn clear_arrivals(&mut self);
```

Comment:
Why is this not removed?

Reply:
It was only used for trimming approvals basing on their arrival time. As ICP approvals are now in stable structures, we do not trim approvals anymore in any ledger.

Reply:
Right, so why do we _keep_ `fn clear_arrivals` in the interface?

Reply:
It is still implemented by `HeapAllowancesData` and is used in post_upgrade to clear the arrivals map in the old deserialized state.